tests/test_xet_utils.py

```diff
@@ -252,3 +253,12 @@ def test_fetch_xet_metadata_with_url_invalid_response(mocker) -> None:
 
     with pytest.raises(ValueError, match="Xet headers have not been correctly set by the server."):
         _fetch_xet_connection_info_with_url(url=url, headers=headers)
+
+
+def test_env_var_hf_hub_disable_xet() -> None:
+    """Test that setting HF_HUB_DISABLE_XET results in is_xet_available() returning False."""
+    from huggingface_hub.utils._runtime import is_xet_available
+
+    os.environ["HF_HUB_DISABLE_XET"] = "1"
+    assert not is_xet_available()
+    del os.environ["HF_HUB_DISABLE_XET"]  # Clean up after the test
```

Comment:
```suggestion
def test_env_var_hf_hub_disable_xet(monkeypatch: pytest.MonkeyPatch) -> None:
    """Test that setting HF_HUB_DISABLE_XET results in is_xet_available() returning False."""
    from huggingface_hub.utils._runtime import is_xet_available
    
    monkeypatch.setenv("HF_HUB_DISABLE_XET", "1")
    assert not is_xet_available()
```
(nit) using `monkeypatch` fixture is usually safer (you are guaranteed that env var is set back even if test fails in the middle of its execution

Reply:
Oh cool, thanks. I didn't realize monkeypatch could be used for env variables. Love it, will change!